### ISSUE TITLE
Cmake .in files are not formatted with `cmake-format`

### ISSUE THREAD

**Issue by JBludau** [kokkos/kokkos#7526]
should be done after open PRs e.g. #7525 that touch the files are merged

**Comment by pzehner** [comment]
This is a noob comment, but

> - Putting all kokkos dependent files of a library into a subdirectory (if you do not expose Kokkos objects in your own interface), and linking with `target_link_libraries(myLib PRIVATE Kokkos::kokkos)`.

subdirectory part aside, isn't this what should be done normally anyway?

**Comment by JBludau** [comment]
> This is a noob comment, but
> 
> > * Putting all kokkos dependent files of a library into a subdirectory (if you do not expose Kokkos objects in your own interface), and linking with `target_link_libraries(myLib PRIVATE Kokkos::kokkos)`.
> 
> subdirectory part aside, isn't this what should be done normally anyway?

yes, you are right. The important thing is the "expose Kokkos objects in your interface". Meaning, you can not have any function that accepts e.g. a `Kokkos::View` in its function interface. So if you do not want to write your own wrapper for some data storage like `View` you can not link `PRIVATE` to `Kokkos::kokkos`. In this case everyone that uses your library will need to compile their code with the compiler and flags that we set (which might not work for them). Now the question is "how do we allow users to only compile files that actually use/expose kokkos" with the compiler we set vs. setting the compiler globally (like we do now). Does this make sense?

**Comment by cedricchevalier19** [comment]
> yes, you are right. The important thing is the "expose Kokkos objects in your interface". Meaning, you can not have any function that accepts e.g. a `Kokkos::View` in its function interface. So if you do not want to write your own wrapper for some data storage like `View` you can not link `PRIVATE` to `Kokkos::kokkos`.

I thought it was the CMake definition `PRIVATE` to not expose anything from underneath. In your case you have also to be `INTERFACE` and CMake rules that `PRIVATE+INTERFACE = PUBLIC`.

**Comment by JBludau** [comment]
> > yes, you are right. The important thing is the "expose Kokkos objects in your interface". Meaning, you can not have any function that accepts e.g. a `Kokkos::View` in its function interface. So if you do not want to write your own wrapper for some data storage like `View` you can not link `PRIVATE` to `Kokkos::kokkos`.
> 
> I thought it was the CMake definition `PRIVATE` to not expose anything from underneath. In your case you have also to be `INTERFACE` and CMake rules that `PRIVATE+INTERFACE = PUBLIC`.

I am not sure if I understand what you want to say. Unfortunately, I used negated statements in my initial comment which was not the best idea

**Comment by cedricchevalier19** [comment]
I was just saying that you are right that you are not able to expose any Kokkos objects in your API if you declare Kokkos as a `PRIVATE` dependency. I think there is nothing specific to Kokkos there.

**Comment by JBludau** [comment]
> I was just saying that you are right that you are not able to expose any Kokkos objects in your API if you declare Kokkos as a `PRIVATE` dependency. I think there is nothing specific to Kokkos there.

yeah. But since Kokkos also adds flags to the linker, we basically infect targets downstream of libs that link `PRIVATE` to us. So we still might need to set how targets that depend on targets that have a `PRIVATE` dependency on Kokkos are linked
